File: gestao_imoveis_app/lib/features/property/domain/entities/inspection_report.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:gestao_imoveis/features/property/domain/value_objects/inspection_type.dart';

part 'inspection_report.freezed.dart';

@freezed
class InspectionReport with _$InspectionReport {
  const factory InspectionReport({
    required int id,
    required int propertyId,
    required InspectionType type,
    required String inspector,
    String? notes,
    required DateTime inspectedAt,
  }) = _InspectionReport;
}

File: API/GestaoImoveisAPI/Domain/Property/InspectionReport.cs
using GestaoImoveisAPI.Domain.Shared;

namespace GestaoImoveisAPI.Domain.Property
{
    public sealed class InspectionReport : Entity
    {
        public int PropertyId { get; private set; }
        public InspectionType Type { get; private set; }
        public string Inspector { get; private set; } = string.Empty;
        public string? Notes { get; private set; }
        public DateTime InspectedAt { get; private set; }

        private InspectionReport() { } // EF Core

        internal static InspectionReport Create(
            int propertyId,
            InspectionType type,
            string inspector,
            string? notes)
        {
            if (string.IsNullOrWhiteSpace(inspector))
                throw new ArgumentException("Nome do vistoriador não pode ser vazio.");

            return new InspectionReport
            {
                PropertyId = propertyId,
                Type = type,
                Inspector = inspector.Trim(),
                Notes = notes?.Trim(),
                InspectedAt = DateTime.UtcNow
            };
        }
    }
}
